test/e2e/scale_test.go

```diff
@@ -0,0 +1,151 @@
+*/
+
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/utils/ptr"
+	capi_e2e "sigs.k8s.io/cluster-api/test/e2e"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	. "sigs.k8s.io/cluster-api/test/framework/ginkgoextensions"
+)
+
+const (
+	scaleClusterControlPlaneEndpointIPPlaceholder   = "scale-cluster-controlplane-endpoint-ip-placeholder"
+	scaleClusterControlPlaneEndpointPortPlaceholder = "scale-cluster-controlplane-endpoint-Port-placeholder"
+	scaleClusterVSphereServerPlaceholder            = "scale-cluster-vsphere-server-placeholder"
+	scaleClusterVSphereTLSThumbprintPlaceholder     = "scale-cluster-vsphere-tls-thumbprint-placeholder"
+	scaleClusterNamePlaceholder                     = "scale-cluster-name-placeholder"
+)
+
+var _ = Describe("When testing the machinery for scale testing using vcsim provider [vcsim] [supervisor] [ClusterClass]", func() {
+	const specName = "quick-start-cluster-class-scale" // prefix must be uniq because this test re-writes the clusterctl config.
+	Setup(specName, func(testSpecificSettingsGetter func() testSettings) {
+		capi_e2e.ScaleSpec(ctx, func() capi_e2e.ScaleSpecInput {
+			// For supporting real environments we would need to properly cleanup the namespaces and namespace specific ip allocations.
+			if testTarget != VCSimTestTarget {
+				Skip("Test should only be run using vcsim provider")
+			}
+
+			return capi_e2e.ScaleSpecInput{
+				E2EConfig:              e2eConfig,
+				ClusterctlConfigPath:   testSpecificSettingsGetter().ClusterctlConfigPath,
+				InfrastructureProvider: ptr.To(clusterctl.DefaultInfrastructureProvider),
+				BootstrapClusterProxy:  bootstrapClusterProxy,
+				ArtifactFolder:         artifactFolder,
+				Flavor:                 ptr.To(testSpecificSettingsGetter().FlavorForMode("topology-runtimesdk")),
+				SkipUpgrade:            true,
+				SkipCleanup:            skipCleanup,
+				ClusterClassName:       getVariableOrFallback(testSpecificSettingsGetter().Variables["CLUSTER_CLASS_NAME"], e2eConfig.GetVariable("CLUSTER_CLASS_NAME")),
+
+				// ClusterCount can be overwritten via `CAPI_SCALE_CLUSTER_COUNT`.
+				ClusterCount: ptr.To[int64](5),
+				// Concurrency can be overwritten via `CAPI_SCALE_CONCURRENCY`.
+				Concurrency: ptr.To[int64](20),
+				// ControlPlaneMachineCount can be overwritten via `CAPI_SCALE_CONTROL_PLANE_MACHINE_COUNT`.
+				ControlPlaneMachineCount: ptr.To[int64](3),
+				// WorkerPerMachineDeploymentCount can be overwritten via `CAPI_SCALE_WORKER_PER_MACHINE_DEPLOYMENT_COUNT`.
+				WorkerPerMachineDeploymentCount: ptr.To[int64](1),
+				// MachineDeploymentCount can be overwritten via `CAPI_SCALE_MACHINE_DEPLOYMENT_COUNT`.
+				MachineDeploymentCount: ptr.To[int64](3),
+				// AdditionalClusterClassCount can be overwritten via `CAPI_SCALE_ADDITIONAL_CLUSTER_CLASS_COUNT`.
+				AdditionalClusterClassCount: ptr.To[int64](4),
```

Comment:
Would it make sense to use the same numbers as in core CAPI per default? (just because this seems very different here)

(differences in the bools below are fine)